instrumentation/logback-1.0/src/main/java/software/amazon/opentelemetry/javaagent/instrumentation/logback_1_0/AwsXrayLoggingEventInstrumentation.java

```diff
@@ -0,0 +1,97 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.instrumentation.logback_1_0;
+
+import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.AgentElementMatchers.implementsInterface;
+import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.ClassLoaderMatcher.hasClassesNamed;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.instrumentation.logback.v1_0.internal.UnionMap;
+import io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Collections;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.bytecode.assign.Assigner.Typing;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public class AwsXrayLoggingEventInstrumentation implements TypeInstrumentation {
+  private static final String TRACE_ID_KEY = "AWS-XRAY-TRACE-ID";
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderOptimization() {
+    return hasClassesNamed("ch.qos.logback.classic.spi.ILoggingEvent");
+  }
+
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    return implementsInterface(named("ch.qos.logback.classic.spi.ILoggingEvent"));
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isMethod()
+            .and(isPublic())
+            .and(named("getMDCPropertyMap").or(named("getMdc")))
+            .and(takesArguments(0)),
+        AwsXrayLoggingEventInstrumentation.class.getName() + "$GetMdcAdvice");
+  }
+
+  public static class GetMdcAdvice {
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void onExit(
+        @Advice.This ILoggingEvent event,
+        @Advice.Return(typing = Typing.DYNAMIC, readOnly = false) Map<String, String> contextData) {
+      if (contextData != null && contextData.containsKey(TRACE_ID_KEY)) {
+        // Assume already instrumented event if traceId is present.
+        return;
+      }
+
+      Span currentSpan = InstrumentationContext.get(ILoggingEvent.class, Span.class).get(event);
+      if (currentSpan == null || !currentSpan.getSpanContext().isValid()) {
+        return;
+      }
+
+      SpanContext spanContext = currentSpan.getSpanContext();
+      String value =
+          TRACE_ID_KEY
+              + ": 1-"
```

Comment:
Originally in the X-Ray SDK, `AWS-XRAY-TRACE-ID` was the default prefix. But customers thought it was unnecessary and verbose, so we made it configurable and disabled by default in the X-Ray agent. I'm not sure how easy it would be to extend the OTel agent configuration in our AWS distro to include such a configuration option, but even if we can't do that I still think we should drop the prefix.

Reply:
So I guess backend features would work without the prefix too (assuming it must if customers are tweaking it)? In that case yeah I'll remove it, the best way to configure the prefix is to add it to the logging config itself.

Reply:
Yep, ServiceLens just searches for the trace ID and doesn't care what's around it (not even the `@spanId` bit, but we should keep that).

Reply:
Cool - removed the prefix